### PR TITLE
trialized event-aligned analysis

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

File: workflow_array_ephys/analysis.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                    'event_start_time', order_by='event_start_time DESC', limit=1)[0]
+            else:
+                continue
+
+            alignment_start_time = (event.Event & key & {'event_type': alignment_spec['start_event_type']}
+                                    & f'event_start_time < {alignment_event_time}')
+            if alignment_start_time:
+                # if there are multiple of such start event, pick the most immediate one prior to the alignment event
+                alignment_start_time = alignment_start_time.fetch(
+                    'event_start_time', order_by='event_start_time DESC', limit=1)[0]
```
Comment (CBroz1) [inline workflow_array_ephys/analysis.py:82]: Smart - I wasn't aware fetch had this functionality

File: tests/__init__.py [hunk 2]
```diff
@@ -53,6 +53,8 @@ def dj_config():
         dj.config.load('./dj_local_conf.json')
     dj.config['safemode'] = False
     dj.config['custom'] = {
+        'ephys_mode': (os.environ.get('EPHYS_MODE')
+                       or dj.config['custom']['ephys_mode']),
```
Comment (CBroz1) [inline tests/__init__.py:57]: I like this approach to navigating the schema choice. I'll make a note to add documentation highlighting this feature to users

**Review by CBroz1 — APPROVED** [review]

**Comment by CBroz1** [comment]
@kabilar - Do you want to take a look before I merge this to the `event` branch? I expect thorough review to take place when merging `event` to `main`

**Comment by kabilar** [comment]
Thanks for merging @CBroz1.  I will review when this work gets merged into `main`.

**Pull request merged**